lang/syn/src/codegen/accounts/constraints.rs

```diff
@@ -320,6 +325,43 @@ pub fn generate_constraint_rent_exempt(
     }
 }
 
+fn generate_constraint_realloc(f: &Field, c: &ConstraintReallocGroup) -> proc_macro2::TokenStream {
+    let field = &f.ident;
+    let new_space = &c.space;
+    let allocator = &c.allocator;
+
+    quote! {
+        let __anchor_rent = Rent::get()?;
+
+        let (__delta_space, __additive) = match #new_space.checked_sub(#field.to_account_info().data_len()) {
+            Some(d) => (d, true),
+            None => (#field.to_account_info().data_len().checked_sub(#new_space).unwrap(), false),
+        };
+
+        if __delta_space > 0 {
+            let __lamport_amt = __anchor_rent.minimum_balance(__delta_space);
```

Comment:
irc, rent is not linear, so it is easier to calculate lamports for both sides and subtract them.

Reply:
the rent per byte is linear and established at genesis, but changed the calculation because more lamports than were necessary were being transferred in/out of the program account.